services/parser-database/tests/test_connector.py

```diff
@@ -71,6 +100,39 @@ def test_get_articles_that_match_keywords_non_empty_result_two_keywords():
     assert result == result_to_assert_4
 
 
+@mock.patch("connector.keywords_in_memory", in_memory_value_mock_no_decimals)
+@mock.patch("connector.articles_in_memory", articles_in_memory)
+def test_get_articles_by_tfidf_value():
+    expected = {
+        "licencia": {"89": {"weight": .3125}, "45": {"weight": .125}, "125": {"weight": .125}},
+        "ciclista": {"5": {"weight": .15625}, "45": {"weight": .25}, "99": {"weight": .3}},
+    }
+    keywords = ["licencia", "ciclista"]
+    with mock.patch("numpy.log", side_effect=logn):
+        assert expected == connector.get_articles_by_tfidf_value(keywords)
+
+
+@mock.patch("connector.keywords_in_memory", in_memory_value_mock_no_decimals)
+@mock.patch("connector.articles_in_memory", articles_in_memory)
+def test_get_articles_by_tfidf_value_empty_result():
+    expected = {
+        "casco": {},
+        "luz": {},
+    }
+    keywords = ["casco", "luz"]
+    with mock.patch("numpy.log", side_effect=logn):
+        assert expected == connector.get_articles_by_tfidf_value(keywords)
+
+
+@mock.patch("connector.keywords_in_memory", in_memory_value_mock_no_decimals)
+@mock.patch("connector.articles_in_memory", articles_in_memory_no_wordCount)
+def test_get_articles_by_tfidf_value_missing_word_count():
+    keywords = ["licencia", "ciclista"]
+    with mock.patch("numpy.log", side_effect=logn):
+        with pytest.raises(KeyError):
+            connector.get_articles_by_tfidf_value(keywords)
```

Comment:
Should. this really be an error? I'm looking through the code and trying to understand why this is a valid behavior. Under what circumstances do you want to return an error rather than an empty result?

Reply:
Well here I'm just saying that if there's no word_count attribute I cannot calculate the tfidf value. I think an error here is more valuable than an empty result for debugging purposes.

Reply:
Ok, I see now. Thanks for explaining.